bdns/dns.go

```diff
@@ -273,10 +272,10 @@ func (dnsClient *impl) exchangeOne(ctx context.Context, hostname string, qtype u
 		case r := <-ch:
 			if r.err != nil {
 				var isRetryable bool
-				// According to the http package documentation, retryable
-				// errors emitted by the http package are of type *url.Error.
-				var urlErr *url.Error
-				isRetryable = errors.As(r.err, &urlErr) && urlErr.Temporary()
+				// Check if the error is a timeout error. Network errors
+				// that can timeout implement the net.Error interface.
+				var netErr net.Error
+				isRetryable = errors.As(r.err, &netErr) && netErr.Timeout()
```

Comment:
Can you explain why this became a `net.Error`? `*url.Error` also offers `.Timeout()`.